spring-integration-core/src/main/java/org/springframework/integration/support/management/graph/GraphServer.java

```diff
@@ -0,0 +1,149 @@
+		Collection<LinkNode> links = new ArrayList<LinkNode>();
+		Map<String, MessageChannelNode> channelInfos = new HashMap<String, MessageChannelNode>();
+		for (Entry<String, MessageChannel> entry : channels.entrySet()) {
+			MessageChannel channel = entry.getValue();
+			MessageChannelNode channelInfo = new MessageChannelNode(entry.getKey(), channel);
+			String beanName = entry.getKey();
+			nodes.add(channelInfo);
+			channelInfos.put(beanName, channelInfo);
+		}
+		for (Entry<String, SourcePollingChannelAdapter> entry : spcas.entrySet()) {
+			SourcePollingChannelAdapter adapter = entry.getValue();
+			MessageSourceNode sourceInfo = sourceInfo(entry.getKey(), adapter);
+			nodes.add(sourceInfo);
+			MessageChannelNode channelInfo = channelInfos.get(sourceInfo.getOutput());
+			if (channelInfo != null) {
+				links.add(new LinkNode(sourceInfo.getNodeId(), channelInfo.getNodeId()));
+			}
+		}
+		for (Entry<String, MessagingGatewaySupport> entry : gateways.entrySet()) {
+			MessagingGatewaySupport gateway = entry.getValue();
+			MessageGatewayNode gatewayInfo = gatewayInfo(entry.getKey(), gateway);
+			nodes.add(gatewayInfo);
+			MessageChannelNode channelInfo = channelInfos.get(gatewayInfo.getOutput());
+			if (channelInfo != null) {
+				links.add(new LinkNode(gatewayInfo.getNodeId(), channelInfo.getNodeId()));
+			}
+		}
+		for (Entry<String, MessageProducerSupport> entry : producers.entrySet()) {
+			MessageProducerSupport producer = entry.getValue();
+			MessageProducerNode producerInfo = producerInfo(entry.getKey(), producer);
+			nodes.add(producerInfo);
+			MessageChannelNode channelInfo = channelInfos.get(producerInfo.getOutput());
+			if (channelInfo != null) {
+				links.add(new LinkNode(producerInfo.getNodeId(), channelInfo.getNodeId()));
+			}
+		}
+		for (Entry<String, IntegrationConsumer> entry : consumers.entrySet()) {
+			IntegrationConsumer consumer = entry.getValue();
+			MessageHandlerNode handlerInfo = handlerInfo(entry.getKey(), consumer);
+			nodes.add(handlerInfo);
+			MessageChannelNode channelInfo = channelInfos.get(handlerInfo.getInput());
+			if (channelInfo != null) {
+				links.add(new LinkNode(channelInfo.getNodeId(), handlerInfo.getNodeId()));
+			}
+			if (handlerInfo.getOutput() != null) {
+				channelInfo = channelInfos.get(handlerInfo.getOutput());
+				if (channelInfo != null) {
+					links.add(new LinkNode(handlerInfo.getNodeId(), channelInfo.getNodeId()));
+				}
+			}
+		}
+		this.graph = new Graph(nodes, links);
+	}
+
+	private MessageGatewayNode gatewayInfo(String name, MessagingGatewaySupport gateway) {
+		return new MessageGatewayNode(name, gateway, gateway.getRequestChannelName());
+	}
+
+	private MessageProducerNode producerInfo(String name, MessageProducerSupport producer) {
+		MessageChannel outputChannel = producer.getOutputChannel();
+		String outputChannelName = outputChannel instanceof NamedComponent
+				? ((NamedComponent) outputChannel).getComponentName() : "__unknown__";
```

Comment:
Can't we use just `outputChannel.toString()` instead?
Seems for me that would be useful to recognize at least the class name there for better traceability...